From d2a492de94297d815c8140356b93b6befffdfde0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 7 Apr 2021 19:38:16 +0200
Subject: [PATCH] data: Add back (hidden) shortcuts for vertical navigation

Users who customized their workspace shortcuts before updating to
GNOME 40 are experiencing troubles when trying to use the same
keybinding for the horizontal shortcuts, because of hidden conflicts
with the vertical ones.

We can address this by registering the old shortcuts with Settings
without showing them in the UI, so that they are taken into account
by Settings' conflict resolution.

https://gitlab.gnome.org/GNOME/mutter/-/issues/1740

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1814>
---
 data/50-mutter-navigation.xml | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/data/50-mutter-navigation.xml b/data/50-mutter-navigation.xml
index b62c636b1a..f76cdb73af 100644
--- a/data/50-mutter-navigation.xml
+++ b/data/50-mutter-navigation.xml
@@ -26,6 +26,14 @@
 	<KeyListEntry name="move-to-workspace-right"
 	              description="Move window one workspace to the right" />
 
+        <KeyListEntry name="move-to-workspace-up"
+                      description="Move window one workspace up"
+                      hidden="true" />
+
+        <KeyListEntry name="move-to-workspace-down"
+                      description="Move window one workspace down"
+                      hidden="true" />
+
 	<KeyListEntry name="move-to-monitor-left"
 	              description="Move window one monitor to the left" />
 
@@ -125,4 +133,12 @@
 	<KeyListEntry name="switch-to-workspace-right"
 	              description="Move to workspace on the right" />
 
+        <KeyListEntry name="switch-to-workspace-up"
+                      description="Move to workspace above"
+                      hidden="true" />
+
+        <KeyListEntry name="switch-to-workspace-down"
+                      description="Move to workspace below"
+                      hidden="true" />
+
 </KeyListEntries>
-- 
GitLab

